app/grandchallenge/evaluation/models.py

```diff
@@ -1695,23 +1702,9 @@ def successful_jobs(self):
             Job.objects.filter(
                 algorithm_image=self.submission.algorithm_image,
                 status=Job.SUCCESS,
-                **extra_filter,
-            )
-            .annotate(
-                inputs_match_count=Count(
-                    "inputs",
-                    filter=Q(
-                        inputs__in={
-                            civ
-                            for civ_set in self.valid_archive_item_values
-                            for civ in civ_set
-                        }
-                    ),
-                ),
-            )
-            .filter(
-                inputs_match_count=self.algorithm_inputs.count(),
+                algorithm_interface__in=self.submission.phase.algorithm_interfaces.all(),
```

Comment:
This is only checking the interfaces and not their values? What happens when jobs are created for that algorithm through the API?

Reply:
Yes, this is not correct. I just realized when adding more tests. It's actually quite complex. I need to return the successful jobs per interface, making sure to check the actual values, and then check that against `valid_archive_item_count_per_interface`.